### ISSUE TITLE
no-inferrable-types: true doesn't error out when doing string concatenation

### ISSUE THREAD

**Issue by pushkin-** [palantir/tslint#3763]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
const msg: string = "Error: " + e.message;
```

with `tslint.json` configuration:

```json
"no-inferrable-types": true
```

#### Actual behavior
I get not tslint error.

#### Expected behavior
Since concatenating something to a string should produce a string in JavaScript, I would expect tslint to complain that the explicit type declaration (`string`) is unnecessary.

Similarly, `const msg: string = "a" + "b";` doesn't give me any tslint error either.

**Comment by JoshuaKGoldberg** [comment]
Seems reasonable that the rule could have an option to allow initial values that are just primitives and operators, as proposed.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
